src/Route.php

```diff
@@ -44,27 +45,20 @@ private function normalizePath(string $path): string
 	 */
 	public function match(Nette\Http\IRequest $httpRequest): ?Request
 	{
-		$path = $this->normalizePath($httpRequest->getUrl()->getPath());
-
 		if (!$this->isHttpMethodSupported($httpRequest->getMethod())) {
 			return NULL;
 		}
 		
-		$route = $this->normalizePath($this->route);
-		$route = str_replace('/', '\/', $route);
+		$path = $httpRequest->getUrl()->getPath();
+		$matches = $this->pathMatcher->match($this->route, $path);
 		
-		// use named subpatterns to match params
-		$routeRegex = preg_replace('/<[\w_-]+>/', '(?$0[\w_-]+)', $route);
-		$routeRegex = '@^' . $routeRegex . '$@';
-
-		$result = preg_match($routeRegex, $path, $matches);
-		if (!$result) {
-			return NULL;
+		if ($matches === null) {
+			return null;
 		}
 		
 		$params = $httpRequest->getQuery();
-		if (is_array($matches) && count($matches) > 1) {
-			$params += array_filter($matches, 'is_string', ARRAY_FILTER_USE_KEY);
+		if ($matches) {
```

Comment:
As i can see, this condition is completely useless as `match()` can only return type of `NULL` or `array`.
In my eyes i would this could be nice, combining lines 59 and 61 together:
```php
$params = array_merge($httpRequest->getQuery(), $matches);
```